### DISCUSSION TITLE
How to learn more about encoding optimizers in deep learning models.

### DISCUSSION THREAD

**Discussion by Jarartur** [facebookresearch/theseus#611] [Q&A] [answered]
Hi all,
I wanted to ask you if you know of any useful materials (books, papers, blogs, youtube, etc.) that I could use to learn about "encoding inductive priors" into the models.
I am not a beginner in neither deep learning nor image processing. I mainly research image registration algorithms, and what I saw in the tutorials here was very interesting. I am not an expert in optimization though, so that's where I am coming from.
I guess what I probably need to understand this concept is to see some more examples, so if you have used or seen theseus in some publicly available code, I surely would appreciate linking it here :).

**Comment by luisenp** [comment, marked as answer]
Hi @Jarartur,

Thanks for your interest in Theseus and also this interesting area of research. I'm not aware of any books covering this topic, but here are some online resources that come to mind:

* Zico Kolter's [Youtube talk](https://www.youtube.com/watch?v=3K9Et7t1XT8) on implicit layers.
* B. Amos and Z. Kolter, OptNet: Differentiable Optimization as a Layer in Neural Networks ([paper](https://proceedings.mlr.press/v70/amos17a/amos17a.pdf)).
* R.T.Q. Chen et al., Neural Ordinary Differential Equations ([paper](https://arxiv.org/pdf/1806.07366.pdf)).
* Our NeurIPS [paper](https://proceedings.neurips.cc/paper_files/paper/2022/file/185969291540b3cd86e70c51e8af5d08-Paper-Conference.pdf), and special attention to the references in the related work and the DNLS subsection (page 3), which list several applications.

Some codebases using Theseus:

* DIPP-Differentiable Integrated Prediction and Planning ([Github](https://github.com/MCZhi/DIPP)).
* SE(3)-Diffusion fields ([website](https://sites.google.com/view/se3dif/home)).
* MidasTouch: Monte-Carlo inference over distributions across sliding touch ([website](https://suddhu.github.io/midastouch-tactile/)).

**Reply by Jarartur** [reply]
Oh, thank you so much for this list :D. I will surely look it all up.
